File: java-src/src/test/java/com/josnelihurt/app/algorithms/hacker_rank/BetweenTwoSetsTest.java
package com.josnelihurt.app.algorithms.hacker_rank;
import java.util.Arrays;
import java.util.List;

import org.junit.Test;

/**
 * Unit test for simple App.
 */
public class BetweenTwoSetsTest 
{
	@Test
	public void TestGetTotalXCase0() {
		List<Integer> a = Arrays.asList(2, 6);
		List<Integer> b = Arrays.asList(24, 36);
		int result = Result.getTotalX(a, b);
		assert result == 2;
	}

	@Test
	public void TestGetTotalXCase8() {
		List<Integer> a = Arrays.asList(3, 4);
		List<Integer> b = Arrays.asList(24, 48);
		int result = Result.getTotalX(a, b);
		assert result == 2;
	}

	@Test
	public void TestGetTotalXCase4() {
		List<Integer> a = Arrays.asList(1);
		List<Integer> b = Arrays.asList(100);
		int result = Result.getTotalX(a, b);
		assert result == 9;
	}
}

File: go-src/algorithms/hacker_rank/between_two_sets_test.go
package hacker_rank

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestGetTotalXCase0(t *testing.T) {
	a := []int32{2, 6}
	b := []int32{24, 36}
	result := getTotalX(a, b)
	assert.Equal(t, int32(2), result)
}
func TestGetTotalXCase8(t *testing.T) {
	a := []int32{3, 4}
	b := []int32{24, 48}
	result := getTotalX(a, b)
	assert.Equal(t, int32(2), result)
}
func TestGetTotalXCase4(t *testing.T) {
	a := []int32{1}
	b := []int32{100}
	result := getTotalX(a, b)
	assert.Equal(t, int32(9), result)
}
